graylog2-web-interface/src/components/users/StartpageFormGroup.tsx

```diff
@@ -0,0 +1,145 @@
+
+const grnId = (grn) => getValuesFromGRN(grn).id;
+const _grnOptionFormatter = ({ id, title }: SharedEntity): Option => ({ value: grnId(id), label: title });
+const typeOptions = [
+  { value: 'dashboard', label: 'Dashboard' },
+  { value: 'stream', label: 'Stream' },
+];
+
+const ADMIN_PERMISSION = '*';
+
+const StartpageFormGroup = ({ userId, permissions }: Props) => {
+  const [dashboards, setDashboards] = useState<Option[] | undefined>();
+  const [streams, setStreams] = useState<Option[] | undefined>();
+
+  useEffect(() => {
+    if (permissions.includes(ADMIN_PERMISSION)) {
+      DashboardsActions.search('', 1, 0).then(({ list }) => setDashboards(list.map(({ id, title }) => ({ value: id, label: title }))));
+
+      StreamsActions.searchPaginated(1, 0, '').then(({ streams: streamsList }) => setStreams(streamsList.map(({ id, title }) => ({ value: id, label: title }))));
+    } else {
+      EntityShareDomain.loadUserSharesPaginated(userId, {
+        ...UNLIMITED_ENTITY_SHARE_REQ,
+        additionalQueries: { entity_type: 'dashboard' },
+      }).then(({ list }) => setDashboards(list.map(_grnOptionFormatter).toArray()))
+        .then(() => EntityShareDomain.loadUserSharesPaginated(userId, {
+          ...UNLIMITED_ENTITY_SHARE_REQ,
+          additionalQueries: { entity_type: 'stream' },
+        }).then(({ list }) => setStreams(list.map(_grnOptionFormatter).toArray())));
+    }
+  }, [permissions, userId]);
+
+  if (!streams || !dashboards) {
+    return <Spinner />;
+  }
+
+  return (
+    <Field name="startpage">
+      {({ field: { name, value, onChange } }) => {
+        const type = value?.type ?? 'dashboard';
+        const options = type === 'dashboard' ? dashboards : streams;
+
+        const error = value?.id && options.findIndex(({ value: v }) => v === value.id) < 0
+          ? <Alert bsStyle="warning">User is missing permission for the configured page</Alert>
+          : null;
+
+        const resetBtn = value?.type
+          ? (
+            <ResetBtn onClick={() => onChange({ target: { name, value: {} } })}>
+              Reset
+            </ResetBtn>
+          )
+          : null;
+
+        return (
+          <>
+            <Input id="startpage"
+                   label="Start page"
+                   help="Select the page the user sees right after log in. Only entities are selectable which the user has permissions for."
+                   labelClassName="col-sm-3"
+                   wrapperClassName="col-sm-9">
+              <>
+                <Container>
+                  <TypeSelect options={typeOptions}
+                              placeholder="Select type"
+                              onChange={(newType) => onChange({ target: { name, value: { type: newType, id: undefined } } })}
+                              value={value?.type} />
+                  <ValueSelect options={options}
+                               placeholder={`Select ${value?.type}`}
```

Comment:
When there is no type selected it says `Select undefined`. 
Let use `Select ${value?.type} ?? 'entity'` here.